blocks/library/gallery/index.js

```diff
@@ -61,16 +70,11 @@ registerBlockType( 'core/gallery', {
 			query( 'div.blocks-gallery figure.blocks-gallery-image img', {
 				url: attr( 'src' ),
 				alt: attr( 'alt' ),
-			} ),
+			} ) || [],
```

Comment:
That was caused by my own confusion and I should’ve removed it. Let me explain, because I think other block implementors can fall in the same trap. I wanted to add a default value for the `columns` attribute. The function that calculates that depends on the number of images and I hoped to avoid the `attributes.images = attributes.images || [];` in `defaultColumnsNumber()`. I hadn’t realized however that the `attributes` property is evaluated only when parsing an existing block and not when creating a new one. So when I got the missing `attributes.images` on creating a new block this was my first line of defense.

I still haven't internalized where data comes from and the overall attributes flow, starting from a user adding a block, through making changes, persisting, loading, `setAttributes`, etc.

Have you considered making the block a class, which has the `attributes` as a member variable and only modify it through `setAttributes` ala React?